webbpsf/constants.py

```diff
@@ -243,3 +243,15 @@
 JWST_INSCRIBED_DIAMETER = 5.47334  # meters. Middle corners of C segments
 
 JWST_TYPICAL_LOS_JITTER_PER_AXIS = 0.0008 # milliarcseconds jitter, 1 sigma per axis. = approx 1 mas rms radial, typically
+
+# Alignment information about instrument internal pupil masks (
+INSTRUMENT_PUPIL_MASK_DEFAULT_POSITIONS = {
```

Comment:
NIRCam will need multiple values that depend on channel (SW or LW pupil wheel) and mask (field position):
 - SWA_MASK210R
 - SWA_MASK335R
 - SWA_MASK430R
 - SWA_MASKSWB
 - SWA_MASKLWB
 - LWA_MASK210R
 - LWA_MASK335R
 - LWA_MASK430R
 - LWA_MASKSWB
 - LWA_MASKLWB

We could conceivably have similar values for Module B, but we currently have zero information on those masks.

Reply:
I thought that physically there is in the SWA wheel and LWA only one each of the bar and round Lyot stops? Such that, for instance, the 210R, 335R, 430R focal plane masks would all be used with the same physical Lyot mask?  And likewise the SWA_MASKSWB and SWA_MASKLWB would similarly use the same  physical Lyot pupil stop?

I had thought that therefore, the Lyot alignment offsets should be independent of focal plane position? What makes that not the case? Is this from field-position-dependent pupil wander within the instrument?

Reply:
Yes, it appears that field-dependent pupil wander changes the best-fit pupil shift values for each of the occulting masks. We found this to be the case during commissioning when trying to find the tuning the pupil wheel clocking to find a good balance between the M335R and M430R masks.

Reply:
OK. This is going to be a minor pain for book-keeping, because it means the pupil alignment depends on the selected image mask (as well as the pupil mask)... but sometimes it's convenient to do coronagraphic calculations with the image mask left out to get an off-axis PSF (i.e. a sim with image_mask=None and pupil_mask=MASKLWB, or similar). For calculations like that, right now there would be no way for webbpsf to know which image mask to use for the pupil   alignment parameter lookup. 

This feels like a minor practical detail of API and implementation, but it'll need at least a little thought for how we want this to work.

Reply:
The pupil shear has negligible impact on the shape of off-axis PSF. The main thing we care about is the morphology of the PSF under the influence of the occulting mask. For on-axis PSFs, the scattered light in the pupil matters (and therefore alignment of the Lyot stop with the telescope pupil), whereas for off-axis PSFs, the pupil is filled. So in the end, we only care about the cases where `image_mask` is set.